service/matching/matcher.go

```diff
@@ -60,14 +62,18 @@ type TaskMatcher struct {
 	// rateLimiter that limits the rate at which tasks can be dispatched to consumers
 	rateLimiter quotas.RateLimiter
 
-	fwdr           *Forwarder
-	metricsHandler metrics.Handler // namespace metric scope
-	numPartitions  func() int      // number of task queue partitions
+	fwdr                   *Forwarder
+	metricsHandler         metrics.Handler // namespace metric scope
+	numPartitions          func() int      // number of task queue partitions
+	backlogTasksCreateTime map[int64]int
+	backlogTasksLock       sync.RWMutex
```

Comment:
my usual approach is to start with a Mutex and only use RWMutex if benchmarking shows a problem, or it's obviously very read-heavy. this seems read-write balanced so I'd just start with a Mutex